raiden/ui/web/src/app/app.component.ts

```diff
@@ -7,17 +10,42 @@ import { SharedService } from './services/shared.service';
     templateUrl: './app.component.html',
     styleUrls: ['./app.component.css'],
 })
-export class AppComponent implements OnInit {
+export class AppComponent implements OnInit, OnDestroy {
     public title = 'Raiden';
     public raidenAddress;
-    public menuCollapsed = false;
 
-    constructor(public sharedService: SharedService,
-                public raidenService: RaidenService) { }
+    pendingRequests = 0;
+    private sub: Subscription;
+
+    constructor(
+        private sharedService: SharedService,
+        private raidenService: RaidenService,
+        private balanceCheckerService: ChannelPollingService
+    ) {
+    }
 
     ngOnInit() {
-        this.raidenService.getRaidenAddress()
-            .subscribe((address) => this.raidenAddress = address);
+        this.raidenService.getRaidenAddress().subscribe((address) => this.raidenAddress = address);
+        this.sub = this.sharedService.pendingRequests.subscribe((pendingRequests) => {
+            setTimeout(() => {
+                this.pendingRequests = pendingRequests;
+            });
+        });
+
+        this.balanceCheckerService.startMonitoring();
```

Comment:
I don't think you need these `startMonitoring`/`stopMonitoring` in the `balanceCheckerService`: a more _Observable-centric_ approach should be to expose a shared `Observable` there (maybe through a public readonly property or getter) directly, containing your logic directly inside the stream, and then subscribing here (in the top level component), possibly discarding any value, the point is just the subscription, so if the observable is subscribed, it acts as `startMonitoring`, and when last subscription is unsubscribed, it'll automatically stop monitoring. And, therefore, if it's just a more complex observable, you may as well just put it inside `RaidenService` directly (or keep it in a separate service too, if you want them separate)

Reply:
Ok I am on it :)